lib/src/insertion_order_parser.dart

```diff
@@ -0,0 +1,193 @@
+  /// instance from [map].
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange
+      _createDateRange(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) {
+      return InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange();
+    }
+
+    final dateRange =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange()
+          ..startDate = _createDate(map['startDate'])
+          ..endDate = _createDate(map['endDate']);
+
+    return dateRange;
+  }
+
+  /// Creates an
+  /// [InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date]
+  /// instance from [map].
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date
+      _createDate(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) {
+      return InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date();
+    }
+
+    final date =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date()
+          ..year = map['year'] ?? emptyEntry
+          ..month = map['month'] ?? emptyEntry
+          ..day = map['day'] ?? emptyEntry;
+
+    return date;
+  }
+
+  /// Creates an [InsertionOrder_EntityStatus] enum from [map].
+  ///
+  /// Returns default value [ENTITY_STATUS_UNSPECIFIED]
+  /// if [target] doesn't correspond to any enum values.
+  static InsertionOrder_EntityStatus _createEntityStatus(String target) {
+    for (var status in InsertionOrder_EntityStatus.values) {
+      if (status.name == target) {
+        return status;
+      }
+    }
+    return InsertionOrder_EntityStatus.ENTITY_STATUS_UNSPECIFIED;
+  }
+
+  /// Creates an [InsertionOrder_Pacing_PacingPeriod] enum from [map].
+  ///
+  /// Returns default value [PACING_PERIOD_UNSPECIFIED]
+  /// if [target] doesn't correspond to any enum values.
+  static InsertionOrder_Pacing_PacingPeriod _createPacingPeriod(String target) {
+    for (var period in InsertionOrder_Pacing_PacingPeriod.values) {
+      if (period.name == target) {
+        return period;
+      }
+    }
+    return InsertionOrder_Pacing_PacingPeriod.PACING_PERIOD_UNSPECIFIED;
+  }
+
+  /// Creates an [InsertionOrder_Pacing_PacingType] enum from [map].
+  ///
+  /// Returns default value [PACING_TYPE_UNSPECIFIED]
+  /// if [target] doesn't correspond to any enum values.
+  static InsertionOrder_Pacing_PacingType _createPacingType(String target) {
+    for (var type in InsertionOrder_Pacing_PacingType.values) {
+      if (type.name == target) {
+        return type;
+      }
+    }
+    return InsertionOrder_Pacing_PacingType.PACING_TYPE_UNSPECIFIED;
+  }
+
+  /// Creates an [InsertionOrder_InsertionOrderBudget_BudgetUnit] from [map].
+  ///
+  /// Returns default value [BUDGET_UNIT_UNSPECIFIED]
+  /// if [target] doesn't correspond to any enum values.
+  static InsertionOrder_InsertionOrderBudget_BudgetUnit _createBudgetUnit(
```

Comment:
is there a better way to parse the string into an enum?
or if not that maybe a static map?

Reply:
aha static map, good point

Reply:
can this method be replaced by something like
_enumMap.get(blah) ?? default val ?
that may be more idiomatic
do you think splitting this out adds to readability?